### ISSUE TITLE
Can not create new project

### ISSUE THREAD

**Issue by chbe8475** [fabric8io/fabric8#6347]
I am running fabric8 on GKE. I used gofabric8 version 0.4.66 to deploy.
I get the console up and running (after setting up the wildcard DNS properly).
When I create a new application by using the console the console gets stuck in an eternal loop.
I choose the Microservice project type, I give the app a name and click next.
I get to the next page which displays three different tabs (Core, Secrets and Pipeline) where the Pipeline tab is selected, and nothing happens. Pressing Ctrl+Shift+Q in firefox shows me the network traffic and it seems like the page is stuck in an eternal loop fetching the same stuff over and over again.

Is this a bug or does this only happen to me?

**Comment by rawlingsj** [comment]
@chbe8475 sounds like something's not quite right.  

Are there any errors in the browser java script console or fabric8 forge pod?

```
kubectl logs fabric8-forge
```

**Comment by chbe8475** [comment]
Lots of errors.

The first error message is:
`Failed to command remote repo http://gogs.default.vagrant.f8/gogsadmin/testet.git due: http://gogs.default.vagrant.f8/gogsadmin/testet.git: cannot open git-upload-pack`

Followed by a very long stack trace, followed by:
`Error occurred during error handling, give up!`
Followed by more stack trace followed by:
`Caused by: 
java.net.UnknownHostException: gogs.default.vagrant.f8`
Followed by some more stack traces.
Then it all ends with lots of rows saying:
`Both io.fabric8.forge.rest.RootResource#ping and io.fabric8.forge.rest.RootResource#ping are equal candidates for handling the current request which can lead to unpredictable results`

The unknown host is strange since I can copy that exact host name and paste it in to my browser and I get to gogs. Also I can find my application in gogs.

**Comment by rawlingsj** [comment]
If I understand your setup correctly from the [other issue](https://github.com/fabric8io/gofabric8/issues/171#issuecomment-246986662) then I think the problem is that the DNS server is running locally so the fabric8-forge pod on GKE can't use that to route to gogs.  

This is an issue that I think we've had before and I cant remember why we need to use the external gogs URL from within the fabric8-forge pod.  @jstrachan dont suppose you can remember?  

Whilst we try and find out if there's anything we can do our side is there any possibility you can use a wildcard DNS that is accessible from within the pod?  For example we use https://www.cloudflare.com/ as its easy and free for this type of use.

Alternatively we go back to the other suggestion in https://github.com/fabric8io/gofabric8/issues/171#issuecomment-246954822

TL;DR as you're using a single node cluster use NodePort
1. edit exposecontroller config
```
kubectl edit configmap exposecontroller
```
2. replace the `expose-rule: ingress` with `expose-rule: node-port`

3. Delete all the pods
```
kubectl delete pods --all
```
the exposecontroller will update the exposeURL annotation on each service

4. open the console at its new url using

```
gofabric8 service fabric8
```

5. try to create another project.

**Comment by rawlingsj** [comment]
Note I've just edited the comment above to add an extra step at #4

**Comment by chbe8475** [comment]
Ok, I did as described above but now I cant reach the console at all from my client. The url is now an ip address and a port number (32305). The ip address is the public ip of my kubernetes cluster in GKE.

**Comment by rawlingsj** [comment]
dam - I wonder if theres a GKE firewall preventing it?  just looking...

**Comment by rawlingsj** [comment]
This is something I've not tried before, I've used loadbalancer types before but kept exceeding quotas so that's why we recommend the ingress approach but as you've found that requires the wildcard DNS to be routable from within the fabric8-forge pod.  I think there's something we can do here to fix fabric8-forge but until then our options are:

1. use a public wildcard dns we suggest https://www.cloudflare.com/
2. continue trying to use nodeport (not something I've seen done on GKE before so we may keep hitting problems)

If we carry on with 2, then we can try to update the GKE firewall to accept a low port range so that the nodeports (32305 and more) work.

to do this I think we'd need to edit the firewall rule thats being used..
```
gcloud compute firewall-rules list
```
and update it for nodeports using the docs https://cloud.google.com/sdk/gcloud/reference/compute/firewall-rules/update

I'm juggling a few different things at the moment so not able to try this myself yet but if you wanted to give it a go that's my suggestion until I can confirm hopefully later this evening.

**Comment by chbe8475** [comment]
I will continue with 2, for a while.

**Comment by chbe8475** [comment]
I opened up the firewall and now I can reach the fabric8 console if I use the public ip of the node. However, fabric8 uses the public ip of the cluster, I can see this by clicking on the 'services' in the console. All services are displayed with the 'wrong' ip.

Also the log in the browser shows me that it is trying to reach the wrong ip address.

Is there somewhere that I can change this ip so that everything point to the correct ip address?

**Comment by chbe8475** [comment]
Ok, I tried another way to get fabric8 running.
I followed [this guide](https://fabric8.io/guide/getStarted/vagrant-kubernetes.html). I am running virtual box 5.0.16 and vagrant 1.8.5.
After I ran `vagrant up` I got several `default: Warning: Authentication failure. Retrying...`. I solved that by using ssh to login to the machine and running `chmod 600 .ssh/authorized_keys`.
I also had to edit the start-kubernetes files in the bin folder and remove a 'shared' flag on volume /var/lib/kubelet.
After a while I got this warning `default: WARNING: IPv4 forwarding is disabled. Networking will not work.` I dont know if that is bad, but probably it is.
After this the `default: switched to context "default".` is displayed and then the installation just sleeps, forever....

Not my day today.
